spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SerializableTableWithSize.java

```diff
@@ -49,6 +61,14 @@ public static Table copyOf(Table table) {
     }
   }
 
+  @Override
+  public void close() throws Exception {
+    if (serializationMarker == null) {
+      LOG.info("Releasing resources");
+      io().close();
+    }
+  }
+
   public static class SerializableMetadataTableWithSize extends SerializableMetadataTable
```

Comment:
Why not do the same cleanup for `MetadataaTable` as well?

Reply:
Good catch, I overlooked it does not extend the main class.